worker/api.js

```diff
@@ -0,0 +1,64 @@
+export async function handleAPIRequest(request) {
+  const url = new URL(request.url);
+
+  if (url.pathname == "/api/versions") {
+    return handleVersionRequest(request);
+  }
+
+  return new Response("API method not found: " + url.pathname, {
+    status: 404,
+    statusText: "Not Found"
+  });
+}
+
+const corsHeaders = {
+  "Access-Control-Allow-Origin": "*",
+  "Access-Control-Allow-Methods": "GET, HEAD, OPTIONS",
+  "Access-Control-Allow-Headers": "Content-Type"
```

Comment:
AFAIK, you don't need to whitelist `Content-Type`, it's a "CORS-safelisted request header" (https://developer.mozilla.org/en-US/docs/Glossary/CORS-safelisted_request_header), i.e. one of the headers that is always allowed.